### ISSUE TITLE
Is it possible to compute gradients of the attributions with respect to sample

### ISSUE THREAD

**Issue by cys515** [meta-pytorch/captum#1221]
Hello, 

My code is as follows:
```python
from captum.attr import Saliency
...... #load test sample, label, and model
sample = sample.requires_grad_(True)  #a Tensor with the size of [1,50,50]
Grad = Saliency(model)
attributions = Grad.attribution(sample,target=label)
attributions = torch.abs(attribution)
min_attr = torch.min(attribution)
max_attr = torch.max(attribution)
attributions = (attributions-min_attr)/(max_attr-min_attr)
y_mesh, x_mesh = torch.meshgrid(torch.arange(50, dtype=torch.float64), torch.arange(50, dtype=torch.float64)) 
mass_center = torch.stack([torch.sum(attributions*x_mesh)/(50*50),torch.sum(attributions*y_mesh)/(50*50)])
mass_center_loss = -torch.sum((mass_center-mass_center0)**2)  #mass_center0 is a consant
temp = -torch.autograd.grad(mass_center_loss, sample)[0]
``` 

I would like to calculate the gradient of mass_center_loss with respect to the input sample,  
but it was an error: One of the differentiated Tensors appears to not have been used in graph.
I guess this might be the reason why the sample doesn't record the gradient in captum fuction.
So what can I do to preserve the gradient information of the samples in the calculation of attribution?

**Comment by craymichael** [comment]
Apologies for the delay.

Most Captum gradient methods compute attributions with `torch.autograd.grad(..., create_graph=False)`. That means the returned attribution tensor is not meant to preserve a higher-order graph for differentiating the attribution again with respect to the input.

For Saliency, you can use a custom gradient function that sets `create_graph=True`. This is the same general workaround discussed in #652:

```python
from captum._utils.common import _run_forward
from captum.attr import Saliency
import torch

def compute_gradients_with_graph(forward_fn, inputs, target_ind=None, additional_forward_args=None):
    with torch.autograd.set_grad_enabled(True):
        outputs = _run_forward(forward_fn, inputs, target_ind, additional_forward_args)
        assert outputs[0].numel() == 1
        return torch.autograd.grad(torch.unbind(outputs), inputs, create_graph=True)

saliency = Saliency(model)
saliency.gradient_func = compute_gradients_with_graph

attributions = saliency.attribute(sample, target=label, abs=False)
```

Two additional points:

- Avoid `torch.abs` if you need smooth higher-order gradients around zero; it is nondifferentiable at zero.
- Do not detach / convert to numpy before computing `mass_center_loss`.

This pattern is more reliable for first-order gradient methods than for methods such as LRP, which use more invasive backward hooks.

This should answer the question, so I'll close this for now. Please feel free to reply with any follow-up questions.
